index.html

```diff
@@ -0,0 +1,191 @@
+        <h3>
+          Extensions to the `Navigator` interface
+        </h3>
+        <pre class="idl">
+          partial interface Navigator {
+            [SecureContext] Promise&lt;TCPSocket&gt; openTCPSocket(optional SocketOptions options = {});
+            [SecureContext] Promise&lt;UDPSocket&gt; openUDPSocket(optional SocketOptions options = {});
+          };
+        </pre>
+        <p>
+          User agents that do not support direct sockets SHOULD NOT expose
+          {{Navigator/openTCPSocket()}} or {{Navigator/openUDPSocket()}} on the
+          {{Navigator}} interface.
+        </p>
+        <section data-dfn-for="SocketOptions">
+          <h3>
+            `SocketOptions` dictionary
+          </h3>
+          <pre class="idl">
+          dictionary SocketOptions {
+            DOMString localAddress;
+            unsigned short localPort;
+
+            DOMString remoteAddress;
+            unsigned short remotePort;
+
+            unsigned long sendBufferSize;
+            unsigned long receiveBufferSize;
+
+            boolean keepAlive;
+            boolean noDelay;
+          };
+        </pre>
+          <p>
+            The <dfn>SocketOptions</dfn> dictionary consists of several
+            optional members:
+          </p>
+          <dl data-sort="">
+            <dt>
+              <dfn>localAddress</dfn> member
+            </dt>
+            <dt>
+              <dfn>localPort</dfn> member
+            </dt>
+            <dt>
+              <dfn>remoteAddress</dfn> member
+            </dt>
+            <dt>
+              <dfn>remotePort</dfn> member
+            </dt>
+            <dt>
+              <dfn>sendBufferSize</dfn> member
+            </dt>
+            <dt>
+              <dfn>receiveBufferSize</dfn> member
+            </dt>
+            <dt>
+              <dfn>keepAlive</dfn> member
+            </dt>
+            <dd>
+              Only relevant for TCP.
+            </dd>
+            <dt>
+              <dfn>noDelay</dfn> member
+            </dt>
+            <dd>
+              Only relevant for TCP.
+            </dd>
+          </dl>
+          <div class="note">
+            The only current implementation (Chromium) ignores
+            {{SocketOptions/localAddress}} and {{SocketOptions/localPort}}.
+          </div>
+        </section>
+        <section data-dfn-for="TCPSocket">
+          <h3>
+            `TCPSocket` interface
+          </h3>
+          <pre class="idl">
+          [Exposed=Window, SecureContext] interface TCPSocket {
+            Promise&lt;undefined&gt; close();
+
+            readonly attribute DOMString remoteAddress;
+            readonly attribute unsigned short remotePort;
+            readonly attribute ReadableStream readable;
+            readonly attribute WritableStream writable;
+          };
+        </pre>
+          <div class="note">
+            The only current implementation (Chromium) does not support "byob"
```

Comment:
Link to BYOB definition? https://streams.spec.whatwg.org/#enumdef-readablestreamreadermode

Reply:
done